### PR TITLE
test(mme): Add further s1ap mme handlers tests

### PR DESCRIPTION
**Pull request by ardzoht**
## Summary

- Similar to #9398, this PR adds tests for uncovered S1AP handlers and procedures
- Refactors some of the repeated code into helper functions
- Adds missing declaration for some of the s1ap_mme_handlers functions

## Test Plan

- `make test_oai`

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_nas_procedures.c [hunk 1]
```diff
@@ -444,17 +445,6 @@ status_code_e s1ap_mme_handle_nas_non_delivery(
       S1ap_NASNonDeliveryIndication_IEs_t, ie, container,
       S1ap_ProtocolIE_ID_id_Cause, true);
 
-  /*
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_nas_procedures.c]: deleted repeated code

**Comment by ulaskozat** [comment]
conflicting file.

File: lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_mme_handlers.cpp [hunk 2]
```diff
@@ -381,5 +372,338 @@ TEST_F(S1apMmeHandlersTest, GenerateUEContextReleaseCommand) {
...
+  // Freeing pdu and payload data
+  bdestroy_wrapper(&payload);
+  ASN_STRUCT_FREE_CONTENTS_ONLY(asn_DEF_S1ap_S1AP_PDU, &pdu);
+  ASN_STRUCT_FREE_CONTENTS_ONLY(asn_DEF_S1ap_S1AP_PDU, &pdu_s1);
+
+  // Sleep to ensure that messages are received and contexts are released
+  std::this_thread::sleep_for(std::chrono::milliseconds(500));
+}
+
+TEST_F(S1apMmeHandlersTest, HandleS1apErabRelCmd) {
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_mme_handlers.cpp:534]: In general I think we should have expect calls for downstream and upstream tasks (e.g., expect to be called zero times). also should not we checking s1ap states at critical parts?

Comment (ardzoht) [reply lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_mme_handlers.cpp:534]: @ulaskozat Added state validation for tests that was missing it, as well as expect calls.

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**